crates/core/src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![feature(async_fn_in_trait)]
```

Comment:
avoid this, there is no good reason to use it and require nightly; where necessary just do `impl Future<Output = _>`